logicrunner/builtin/contract/deposit/deposit.go

```diff
@@ -131,17 +129,24 @@ func (d *Deposit) Confirm(migrationDaemonRef string, txHash string, amountStr st
 
 // Check amount field in confirmation from migration daemons.
 func (d *Deposit) checkAmount(activeDaemons []string) error {
-	if len(activeDaemons) > 0 {
-		amount := d.MigrationDaemonConfirms[activeDaemons[0]]
-		for i := 0; i < insolar.GenesisAmountActiveMigrationDaemonMembers; i++ {
-			if amount != d.MigrationDaemonConfirms[activeDaemons[i]] {
+	if activeDaemons == nil || len(activeDaemons) == 0 {
+		return fmt.Errorf(" list with migration daemons member is empty ")
+	}
+	result := ""
+	for _, migrationRef := range activeDaemons {
+		if amount, ok := d.MigrationDaemonConfirms[migrationRef]; ok {
+			if result == "" {
+				result = amount
+				continue
+			}
+			if result != d.MigrationDaemonConfirms[migrationRef] {
```

Comment:
Why not just use `amount ` instead of `d.MigrationDaemonConfirms[migrationRef]?`

Reply:
Done